packages/neuron-wallet/src/startup/sync-block-task/indexer.ts

```diff
@@ -0,0 +1,55 @@
+import { remote } from 'electron'
+import AddressService from 'services/addresses'
+import LockUtils from 'models/lock-utils'
+import IndexerQueue from 'services/indexer/queue'
+
+import { initDatabase } from './init-database'
+
+const { nodeService, addressDbChangedSubject, walletCreatedSubject } = remote.require(
+  './startup/sync-block-task/params'
+)
+
+// maybe should call this every time when new address generated
+// load all addresses and convert to lockHashes
+export const loadAddressesAndConvert = async (): Promise<string[]> => {
+  const addresses: string[] = (await AddressService.allAddresses()).map(addr => addr.address)
+  const lockHashes: string[] = await LockUtils.addressesToAllLockHashes(addresses)
+  return lockHashes
+}
+
+// call this after network switched
+let indexerQueue: IndexerQueue | undefined
+export const switchNetwork = async (nodeURL: string) => {
+  // stop all blocks service
+  if (indexerQueue) {
```

Comment:
Does the `indexerQueue` have a value when setup?

Reply:
When you call switch network multi times.